src/components/Personalization/dom-actions/setStyles.js

```diff
@@ -12,10 +12,15 @@ governing permissions and limitations under the License.
 
 import { setStyle } from "./dom/index.js";
 
-export default (container, styles) => {
-  const { priority, ...style } = styles;
+export default (container, styles, decorateProposition) => {
+  return new Promise((resolve) => {
+    const { priority, ...style } = styles;
 
-  Object.keys(style).forEach((key) => {
-    setStyle(container, key, style[key], priority);
+    Object.keys(style).forEach((key) => {
+      decorateProposition(container);
```

Comment:
The container is the same, no need to re-decorate it.

Reply:
good catch, thank you! 🙏